### PR TITLE
SSO: Install Keycloak & LDAP

### LINKED ISSUE #4: Do we want to use SSO?
Do we want to use a Single-Sign-On tool for our services? Currently it does not matter which tool we would use.

Please tell us what you think of the idea. Here are some pros and cons I already got:
**Pro:**
* Easy method to get access to services (no need to register with each one and might need different passwords)
  * Easier for users to use a new service

**Con:**
* People are needed to maintain the User-Database (e.g. create new users, delete old, give permissions...)
* Maybe it's not possible to have different e-mail addresses with services
* Difficult for user's who only want an account with one service
* Train user to enter their chaos-jetzt "master-password" at pages who might not use the SSO

**Open questions**
* Is it possible to have different e-mail addresses with services?
* What should happen with already existing accounts?

### PR DESCRIPTION
**Pull request by n0emis**
This PR installs Keycloak and slapd for usage as a SSO.

Closes #4

### CODE REVIEW

**Comment by n0emis** [comment]
This is already deployed on dev at https://sso.dev.chaos.jetzt

**Review by leona-ya — APPROVED** [review]

File: playbooks/site.yml [hunk 1]
```diff
@@ -62,6 +62,25 @@
   tags:
     - traefik
 
+- hosts: web
+  roles:
+    - geerlingguy.postgresql
+  tags: pretix,matrix,keycloak
```
Comment (e1mo) [inline playbooks/site.yml]:
```suggestion
  tags:
    - pretix
    - matrix
    - keycloak
```

File: templates/traefik/dynamic/keycloak.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+http:
+  routers:
+    keycloak-http:
+      rule: "Host(`sso.{{ base_url }}`)"
```
Comment (e1mo) [inline templates/traefik/dynamic/keycloak.yml]:
```suggestion
      rule: "Host(`{{ keycloak_url }}`)"
```

File: templates/traefik/dynamic/keycloak.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      service: "keycloak@file"
+      entryPoints:
+        - "http"
+      middlewares:
+        - "https_redirect"
+    keycloak-https:
+      rule: "Host(`sso.{{ base_url }}`)"
```
Comment (e1mo) [inline templates/traefik/dynamic/keycloak.yml]:
```suggestion
      rule: "Host(`{{ keycloak_url }}`)"
```

File: group_vars/all/keycloak.yml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+keycloak_postgresql_username: "{{ lookup('passwordstore', 'infra/keycloak/db subkey=user') }}"
+keycloak_postgresql_password: "{{ lookup('passwordstore', 'infra/keycloak/db') }}"
+keycloak_postgresql_database: "keycloak"
+
+keycloak_configure_firewall: false
+keycloak_http_port: 8081
```
Comment (e1mo) [inline group_vars/all/keycloak.yml:7]:
```suggestion
keycloak_url: "sso.{{ base_url }}"
keycloak_http_port: 8081
```

**Review by e1mo — CHANGES_REQUESTED** [review]

**Review by e1mo — APPROVED** [review]

**Pull request merged**